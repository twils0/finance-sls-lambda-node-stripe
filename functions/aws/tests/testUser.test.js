const AWS = require('aws-sdk'); // eslint-disable-line

const { testUser } = require('../testUser');

const adminGetUser = jest.fn();

AWS.CognitoIdentityServiceProvider = jest.fn(() => ({
  adminGetUser,
}));

const email = 'test@test.com';
const userId = 'testUserID';

describe('functions', () => {
  describe('aws', () => {
    describe('testUser', () => {
      afterEach(() => {
        adminGetUser.mockReset();
      });

      it("correctly resolves adminGetUser when a 'UserNotFoundException' error is thrown", async () => {
        const error = {
          code: 'UserNotFoundException',
          message: 'testMessage',
        };
        adminGetUser.mockImplementation((params, callback) => callback(error));

        const result = await testUser(email);

        expect(result).toEqual(null);
      });

      it('fails and returns callback error', async () => {
        const error = {
          code: 'testCode',
          message: 'testMessage',
        };
        adminGetUser.mockImplementation((params, callback) => callback(error));

        try {
          await testUser(email);
        } catch (errorCatch) {
          expect(errorCatch).toEqual(error);
        }
      });

      it("fails and returns a 'UsernameExistsException' error when not provided a userId", async () => {
        const response = {
          UserAttributes: [
            {
              Name: 'testName',
              Value: 'testValue',
            },
          ],
        };
        adminGetUser.mockImplementation((params, callback) => callback(null, response));

        try {
          await testUser(email);
        } catch (errorCatch) {
          expect(errorCatch).toMatchSnapshot();
        }
      });

      it("fails and returns a 'UsernameExistsException' error when provided userId that does not match", async () => {
        const response = {
          UserAttributes: [
            {
              Name: 'custom:stripe_customer_id',
              Value: 'testValue',
            },
          ],
        };
        adminGetUser.mockImplementation((params, callback) => callback(null, response));

        try {
          await testUser(email);
        } catch (errorCatch) {
          expect(errorCatch).toMatchSnapshot();
        }
      });

      it('correctly resolves adminGetUser when provided userId that matches', async () => {
        const response = {
          UserAttributes: [
            {
              Name: 'custom:stripe_customer_id',
              Value: userId,
            },
          ],
        };
        adminGetUser.mockImplementation((params, callback) => callback(null, response));

        const result = await testUser(email, userId);

        expect(result).toEqual(null);
      });
    });
  });
});
