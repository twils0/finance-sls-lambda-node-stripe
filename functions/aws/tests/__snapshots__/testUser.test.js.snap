// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`functions aws testUser fails and returns a 'UsernameExistsException' error when not provided a userId 1`] = `
Object {
  "code": "UsernameExistsException",
  "message": "This email is already in use",
}
`;

exports[`functions aws testUser fails and returns a 'UsernameExistsException' error when provided userId that does not match 1`] = `
Object {
  "code": "UsernameExistsException",
  "message": "This email is already in use",
}
`;
